source/patterns/@aws-solutions-constructs/aws-fargate-ssmstringparameter/test/fargate-ssmstringparameter.test.ts

```diff
@@ -0,0 +1,758 @@
+              },
+              "/fake-repo:latest"
+            ]
+          ]
+        },
+        MemoryReservation: 512,
+        Name: "test-container",
+        PortMappings: [
+          {
+            ContainerPort: 8080,
+            Protocol: "tcp"
+          }
+        ]
+      }
+    ]
+  });
+
+  expect(stack).toHaveResourceLike("AWS::SSM::Parameter", {
+    Name: "FooParameter",
+    Value: "Foo"
+  });
+  expect(stack).toHaveResourceLike("AWS::EC2::VPC", {
+    CidrBlock: '172.168.0.0/16'
+  });
+
+  expect(stack).toHaveResourceLike("AWS::IAM::Policy", {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            "ssm:DescribeParameters",
+            "ssm:GetParameters",
+            "ssm:GetParameter",
+            "ssm:GetParameterHistory"
+          ],
+          Effect: "Allow",
+          Resource: {
+            "Fn::Join": [
+              "",
+              [
+                "arn:",
+                {
+                  Ref: "AWS::Partition"
+                },
+                ":ssm:",
+                {
+                  Ref: "AWS::Region"
+                },
+                ":",
+                {
+                  Ref: "AWS::AccountId"
+                },
+                ":parameter/",
+                {
+                  Ref: "Parameter9E1B4FBA"
+                }
+              ]
+            ]
+          }
+        }
+      ]
+    }
+  });
+
+  expect(stack).toHaveResource("AWS::EC2::VPCEndpoint", {
+    ServiceName: {
+      "Fn::Join": [
+        "",
+        [
+          "com.amazonaws.",
+          {
+            Ref: "AWS::Region"
+          },
+          ".ssm"
+        ]
+      ]
+    }
+  });
+
+  // Confirm we created an Isolated VPC
+  expect(stack).not.toHaveResourceLike('AWS::EC2::InternetGateway', {});
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::ECS::Service', 1);
+  expect(stack).toCountResources('AWS::SSM::Parameter', 1);
+});
+
+test('Test error invalid parameter store permission', () => {
+  const stack = new cdk.Stack();
+  const publicApi = false;
+  const serviceName = 'custom-name';
+
+  const existingVpc = defaults.getTestVpc(stack, publicApi);
+
+  const [testService, testContainer] = defaults.CreateFargateService(stack,
+    'test',
+    existingVpc,
+    undefined,
+    defaults.fakeEcrRepoArn,
+    undefined,
+    undefined,
+    undefined,
+    { serviceName });
+
+  const existingStringParameterObj = createSsmParameterStore(stack);
+
+  const app = () => {
+    new FargateToSsmstringparameter(stack, 'test-construct', {
+      publicApi,
+      existingFargateServiceObject: testService,
+      existingContainerDefinitionObject: testContainer,
+      existingVpc,
+      stringParameterPermissions: 'reed',
+      existingStringParameterObj
+    });
+  };
+
+  expect(app).toThrowError('Invalid stringParameterPermissions submitted - REED');
+});
+
+test('Test error invalid parameter store permission', () => {
```

Comment:
Incorrect test title - this is checks for string parameter ( this title is cut/pasted from prior test?)